lemur/certificates/cli.py

```diff
@@ -775,3 +777,146 @@ def deactivate_entrust_certificates():
+                             )
+
+    log_data["message"] = "Summary of task run"
+    log_data["unique_cert_prefix_count"] = len(unique_prefix)
+    log_data["rotation_disabled_cert_count"] = len(rotation_disabled_certs)
+    log_data["certificate_with_no_change_count"] = len(skipped_certs)
+    log_data["failed_to_determine_if_duplicate_count"] = len(failed_certs)
+
+    current_app.logger.info(log_data)
+
+
+def process_duplicates(duplicate_candidate_cert, skipped_certs, rotation_disabled_certs, unique_prefix, commit):
+    """
+    Process duplicates with same prefix as duplicate_candidate_cert
+
+    :param duplicate_candidate_cert: Name of the certificate which has duplicates
+    :param skipped_certs: List of certificates which will continue to have rotation on (no change)
+    :param rotation_disabled_certs: List of certificates for which rotation got disabled as part of this job
+    :param unique_prefix: List of uniq prefixes to avoid rework
+    :return: Success - True or False; If False, set of duplicates which were not processed
+    """
+    name_without_serial_num = duplicate_candidate_cert.name[:duplicate_candidate_cert.name.rindex("-")]
+    if name_without_serial_num in unique_prefix:
+        return True, None
+
+    unique_prefix.append(name_without_serial_num)
+
+    prefix_to_match = name_without_serial_num + '%'
+    certs_with_same_prefix = get_certificates_with_same_prefix_with_rotate_on(prefix_to_match)
+
+    if len(certs_with_same_prefix) == 1:
+        # this is the only cert with rotation ON, no further action needed
+        skipped_certs.append(certs_with_same_prefix[0].name)
+        metrics.send("disable_rotation_duplicates", "counter", 1,
+                     metric_tags={"status": "skipped", "certificate": certs_with_same_prefix[0].name}
+                     )
+        return True, None
+
+    skip_cert = False
+
+    for matching_cert in certs_with_same_prefix:
+        if matching_cert.name == name_without_serial_num:
+            # There exists a cert with name same as the prefix (most likely there will always be one)
+            # Keep auto rotate on for this one if no cert has endpoint associated
+            fallback_cert_to_rotate = name_without_serial_num
+
+        if matching_cert.name == duplicate_candidate_cert.name:
+            # Same cert, no need to compare
+            continue
+
+        # Even if one of the cert with same prefix has different details, skip this set of certs
+        # it's safe to do so and this logic can be revisited
+        if not is_duplicate(matching_cert, duplicate_candidate_cert):
+            skip_cert = True
+
+    if skip_cert:
+        return False, certs_with_same_prefix
+
+    # Find certs with endpoint, auto-rotate needs to be on for these
+    certs_with_endpoint = []
+    for matching_cert in certs_with_same_prefix:
+        if matching_cert.endpoints:
+            certs_with_endpoint.append(matching_cert.name)
+
+    # If no cert has endpoint
```

Comment:
```
# If no cert has endpoint
```

to 

```
# If no certificate has an endpoint, we will add one to the `certs_with_endpoint`, to allow one certificate to auto-rotate.
```
also maybe changing `certs_with_endpoint` to `certs_to_stay_on_autorotate` would help with context .